### PR TITLE
Strong-name sign the assemblies

### LINKED ISSUE #78: Make the ServiceBusAttachment.Plugin.dll strongly named
---
name: Feature request
about: Suggest an idea for this project

If a dll is not strongly named, it can't be used from other strongly named components. For that, it would be helpful to make the ServiceBusAttachment.Plugin strongly named. There are several benefits by making a dll strongly named as described here 
https://docs.microsoft.com/en-us/dotnet/framework/app-domains/strong-named-assemblies

---

**Is your feature request related to a problem? Please describe.**
A clear and concise description of what the problem is. 
Example. I'm am trying to do [...] but [...]
I am trying to consume the library. However, other components of the library are strongly named, so it gives error.

**Describe the solution you'd like**
Here are the steps to sign an assembly with a strong name
https://docs.microsoft.com/en-us/dotnet/framework/app-domains/how-to-sign-an-assembly-with-a-strong-name

**Describe alternatives you've considered**
There are few alternatives:
1) Sign the dll and consume it from a static source
2) Consume the dll dynamically without signing.
3) Change the consumer to be unsigned (but that's hard as there are chain dependencies). 
**Additional context**

### PR DESCRIPTION
**Pull request by nayanshah**
Fixes #78

### CODE REVIEW

**Comment by SeanFeldman** [comment]
:x: [Build ServiceBus.AttachmentPlugin 0.0.311-preview failed](https://ci.appveyor.com/project/seanfeldman/servicebus-attachmentplugin/builds/21370321) (commit https://github.com/SeanFeldman/ServiceBus.AttachmentPlugin/commit/2b0fdc08e7 by @nashah)

**Comment by SeanFeldman** [comment]
:x: [Build ServiceBus.AttachmentPlugin 0.0.312-preview failed](https://ci.appveyor.com/project/seanfeldman/servicebus-attachmentplugin/builds/21370421) (commit https://github.com/SeanFeldman/ServiceBus.AttachmentPlugin/commit/4d0989ffff by @nashah)

**Comment by SeanFeldman** [comment]
:white_check_mark: [Build ServiceBus.AttachmentPlugin 0.0.321-preview completed](https://ci.appveyor.com/project/seanfeldman/servicebus-attachmentplugin/builds/21376081) (commit https://github.com/SeanFeldman/ServiceBus.AttachmentPlugin/commit/95e3666143 by @nashah)

File: src/ServiceBus.AttachmentPlugin.Tests/ApprovalFiles/ApiApprovals.AzureStorageAttachmentPlugin.approved.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-[assembly: System.Runtime.CompilerServices.InternalsVisibleToAttribute("ServiceBus.AttachmentPlugin.Tests")]
+[assembly: System.Runtime.CompilerServices.InternalsVisibleToAttribute("ServiceBus.AttachmentPlugin.Tests, PublicKey=0024000004800000940000000602000000240000525341310004000001000100b5a2fefe7a1f74bb526125b1630af3f5e30b3961640926ea3b20cdb9a803487af798b02fd52a8d265c3fe7174027dd3f738f0e24213cb26b21189ada2d85d74a3cd2c944c7e57c76c5cba3aa45928fc3c300f9744cfaab92c24ac98d15ec2fe05c0500121be04644881448dd06f15f5cd5a4db3dbb8e9d78e63436045bbe8db5")]
```
[the remark this answers is not in the document]
Comment (SeanFeldman) [reply src/ServiceBus.AttachmentPlugin.Tests/ApprovalFiles/ApiApprovals.AzureStorageAttachmentPlugin.approved.txt]: Public API Approver [test](https://github.com/SeanFeldman/ServiceBus.AttachmentPlugin/blob/develop/src/ServiceBus.AttachmentPlugin.Tests/ApiApprovals.cs).
You need to run it to create changes and modify `.approved.txt` with changes from `.received.txt`.

[the remark this answers is not in the document]
Comment (nayanshah) [reply src/ServiceBus.AttachmentPlugin.Tests/ApprovalFiles/ApiApprovals.AzureStorageAttachmentPlugin.approved.txt]: Good to know, thanks. Running the test produces no changes now.

**Comment by SeanFeldman** [comment]
:white_check_mark: [Build ServiceBus.AttachmentPlugin 0.0.328-preview completed](https://ci.appveyor.com/project/seanfeldman/servicebus-attachmentplugin/builds/21391781) (commit https://github.com/SeanFeldman/ServiceBus.AttachmentPlugin/commit/95e3666143 by @nashah)

**Comment by SeanFeldman** [comment]
:white_check_mark: [Build ServiceBus.AttachmentPlugin 0.0.329-preview completed](https://ci.appveyor.com/project/seanfeldman/servicebus-attachmentplugin/builds/21392251) (commit https://github.com/SeanFeldman/ServiceBus.AttachmentPlugin/commit/8f2977ad35 by @nashah)

**Comment by SeanFeldman** [comment]
:white_check_mark: [Build ServiceBus.AttachmentPlugin 0.0.401-preview completed](https://ci.appveyor.com/project/seanfeldman/servicebus-attachmentplugin/builds/21668015) (commit https://github.com/SeanFeldman/ServiceBus.AttachmentPlugin/commit/044f198bcb by @nashah)

**Comment by SeanFeldman** [comment]
:white_check_mark: [Build ServiceBus.AttachmentPlugin 0.0.402-preview completed](https://ci.appveyor.com/project/seanfeldman/servicebus-attachmentplugin/builds/22181912) (commit https://github.com/SeanFeldman/ServiceBus.AttachmentPlugin/commit/598759e7e8 by @nashah)

**Comment by SeanFeldman** [comment]
:white_check_mark: [Build ServiceBus.AttachmentPlugin 0.0.436-preview completed](https://ci.appveyor.com/project/seanfeldman/servicebus-attachmentplugin/builds/23112693) (commit https://github.com/SeanFeldman/ServiceBus.AttachmentPlugin/commit/8cceb9c17f by @nashah)

**Comment by SeanFeldman** [comment]
@nayanshah this change will be released with 5.0.0 which will be coming out shortly.

**Comment by SeanFeldman** [comment]
@nayanshah could you please share your twitter account with me to give you the credit? Thanks

**Comment by nashah** [comment]
Thanks, Twitter handle is same as here: nayanshah

Sent from Outlook mobile

________________________________

**Pull request merged**